trie/trie_reader.go

```diff
@@ -46,10 +39,16 @@ type trieReader struct {
 }
 
 // newTrieReader initializes the trie reader with the given node reader.
-func newTrieReader(stateRoot, owner common.Hash, db NodeReader) (*trieReader, error) {
-	reader := db.Reader(stateRoot)
-	if reader == nil {
-		return nil, fmt.Errorf("state not found #%x", stateRoot)
+func newTrieReader(stateRoot, owner common.Hash, db *Database) (*trieReader, error) {
+	if stateRoot == (common.Hash{}) || stateRoot == types.EmptyRootHash {
+		if stateRoot == (common.Hash{}) {
+			log.Warn("Initialize trie with empty root")
```

Comment:
Please make a more descriptive error message. I don't really understand what this warning is about ...?

Reply:
I will change it to `log.Error("Zero state root hash!")`. We suppose to use `types.EmptyRootHash` instead of `common.Hash{}` to represent an empty hash. So just add a warning or error which is not supposed to be printed.